Sources/Paywalls/Components/PaywallStackComponent.swift

```diff
@@ -18,10 +18,16 @@ import Foundation
 
 public extension PaywallComponent {
 
-    struct StackComponent: PaywallComponentBase, StackableComponent {
+    // The most common type of stack that can contain any type of PaywallComponent
+    typealias StackComponent = GenericStackComponent<PaywallComponent>
+
+    // A specialized stack that can only hold PaywallComponent.PackageComponent
+    typealias PackagesStackComponent = GenericStackComponent<PaywallComponent.PackageComponent>
+
+    struct GenericStackComponent<T: StackableComponent>: PaywallComponentBase {
```

Comment:
This is _kind of_ the weirdest change 🤷‍♂️ 

**Problem:** Stacks can contain any type components but Package Group needs to only contain Package components.

**Solution:** The main thing is now called `GenericStackComponent` (ewwww). However, there are two typealiases. `StackComponent` is exactly the behavior it was previously which was a stack that took any `PaywallComponent`. The new one is `PackagesStackComponent` which is only used by the package group and it will only accept packages.

Reply:
I actually think I kinda like it! Nice solution that allows us to keep benefiting from compiler type checks whenever we want.